src/System.Text.Encoding.CodePages/src/System/Text/SBCSCodePageEncoding.cs

```diff
@@ -37,6 +37,27 @@ public SBCSCodePageEncoding(int codePage, int dataCodePage) : base(codePage, dat
         {
         }
 
+        private static unsafe void ZeroMemAligned(byte* buffer, int count)
+        {
+            long* pLong = (long*)buffer;
+            long* pLongEnd = (long*)(buffer + count - sizeof(long));
+
+            while (pLong < pLongEnd)
+            {
+                *pLong = 0;
+                pLong++;
+            }
+
+            byte* pByte = (byte*)pLong;
+            byte* pEnd = buffer + count;
+
+            while (pByte < pEnd)
+            {
+                *pByte = 0;
+                pByte++;
+            }
+        }
+
```

Comment:
The implementation looks good to me as we discussed it offline. just put some comment that the method assume the memory pointer is aligned so no one else in the future can use it in wrong way.  
Also it will be good if we can test that this in ARM device if possible.